lib/db/set_test.go

```diff
@@ -659,13 +659,13 @@ func TestCommitted(t *testing.T) {
 		t.Errorf("committed data didn't increase; %d <= %d", c1, c0)
 	}
 
-	// Updating with something identical should not do anything
+	// Updating with something identical should also increase the counter
 
 	s.Update(protocol.LocalDeviceID, local)
 
 	c2 := ldb.Committed()
-	if c2 > c1 {
-		t.Errorf("replace with same contents should do nothing but %d > %d", c2, c1)
+	if c2 <= c1 {
+		t.Errorf("replace with same contents should do increase the counter but %d != %d", c2, c1)
```

Comment:
Broken sentence structure and this test is now mostly useless; I'd delete everything after line 661. This used to test the if statement you removed, because historically we sent unnecessary updates for ignored files or something and these were filtered out by that one.